### PR TITLE
Optionalizing feature of formatting Markdown tables

### LINKED ISSUE #1697: Improvement: テーブル編集時の自動整形を一時的にでも無効にしたい
Environment
------------

### Host

| item     | version |
| ---      | --- |
|OS        | Ubuntu 18.04.4 LTS|
|GROWI     | 3.6.6 |
|node.js   | 10.19.0|
|npm       |6.13.4|
|Using Docker|no|

### Client

| item     | version |
| ---      | --- |
|OS        |macOS 10.14.6|
|browser   |80.0.3987.132|

How to reproduce? (再現手順)
---------------------------
1. テーブルの行を改行する

What happens? (症状)
---------------------
1. markdownの該当テーブル全体が自動整形される

What is the expected result? (期待される動作)
-------------------------------------------
- 例えばCtrl+Iといったコマンドを入力した時だけ、選択範囲のみ整形されるようになってほしい

Note
----
- テーブルの途中からテーブルを分割したい時など、Enterを押すと空白行が作られるため、何もない行を作るのが少し大変に感じています。
- 自動整形は、セルの中で改行していたり、長い文字列が入っている場合など整形の結果テーブルが見にくく、編集しにくくなって不便なケースもあると感じています。
- デフォルトの挙動も便利です。でもオプションで一時的でも変更できればもっと嬉しいです。

### PR DESCRIPTION
**Pull request by ryu-sato**
This PR optionalize feature of formatting Markdown tables.
resolve #1697

The settings are stored in local storage and so the default is false. (Currently, formatting is always executed).

formatting_markdown

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -213,7 +213,7 @@
     "markdown-it-plantuml": "^1.3.0",
     "markdown-it-task-checkbox": "^1.0.6",
     "markdown-it-toc-and-anchor-with-slugid": "^1.1.4",
-    "markdown-table": "^1.1.1",
+    "markdown-table": "^1.1.0",
```
Comment (ryu-sato) [inline package.json:216]: Changed the dependency to be the same as that of "remark-stringify".

File: resource/locales/zh_CN/translation.json [hunk 2]
```diff
@@ -286,6 +286,7 @@
 	},
 	"page_edit": {
 		"Show active line": "显示活动行",
+		"formatting_a_markdown_table": "格式化一个Markdown表",
```
Comment (ryu-sato) [inline resource/locales/zh_CN/translation.json:289]: Automatic translation.

File: src/client/js/components/PageEditor/OptionsSelector.jsx [hunk 3]
```diff
@@ -97,6 +98,17 @@ class OptionsSelector extends React.Component {
     editorContainer.saveOptsToLocalStorage();
   }
 
+  onClickFormattingMarkdownTable(event) {
+    const { editorContainer } = this.props;
+
+    const newValue = !editorContainer.state.editorOptions.formattingMarkdownTable;
+    const newOpts = Object.assign(editorContainer.state.editorOptions, { formattingMarkdownTable: newValue });
```
Comment (yuki-takei) [inline src/client/js/components/PageEditor/OptionsSelector.jsx:105]: Change 'formattingMarkdownTable' to 'ignoreMarkdownTableAutoFormatting'.
As a result, auto-formatting would be enabled in default.

File: resource/locales/en_US/translation.json [hunk 4]
```diff
@@ -308,6 +308,7 @@
   },
   "page_edit": {
     "Show active line": "Show active line",
+    "formatting_a_markdown_table": "Formatting a Markdown Table",
```
Comment (yuki-takei) [inline resource/locales/en_US/translation.json:311]: Change it to "Auto format table"

**Review by yuki-takei — CHANGES_REQUESTED** [review]

File: src/client/js/components/PageEditor/CodeMirrorEditor.jsx [hunk 5]
```diff
@@ -823,6 +824,10 @@ export default class CodeMirrorEditor extends AbstractEditor {
 
     const placeholder = this.state.isGfmMode ? 'Input with Markdown..' : 'Input with Plane Text..';
 
+    const markdownTableOption = {
+      pad: this.props.editorOptions.formattingMarkdownTable === true,
```
[the remark this answers is not in the document]
Comment (ryu-sato) [reply src/client/js/components/PageEditor/CodeMirrorEditor.jsx:828]: You right. I'll modified.

File: src/client/js/components/PageEditor/OptionsSelector.jsx (hunk 3 above)
Comment (ryu-sato) [reply src/client/js/components/PageEditor/OptionsSelector.jsx]: I think this is a good idea and user-friendly.
I will respond.

**Comment by ryu-sato** [comment]
After corresponding FB.

formatting_markdown

**Pull request merged**